server/src/routes/user.js

```diff
@@ -141,13 +141,29 @@ router.get('/:username/services', (req, res) => {
 /* DELETE user(s). */
 router.delete('/:username', (req, res) => {
     if (Object.prototype.hasOwnProperty.call(req.body, 'password')) {
-        if (verifyPassword(req.params.username, req.body.password)) {
-            User.remove({ username: req.params.username }, err => {
-                if (err) throw err;
-                res.status(200);
+        verifyPassword(req.params.username, req.body.password)
+            .then(isMatch => {
+                if (isMatch) {
+                    User.findOneAndRemove({ username: req.params.username })
+                        .then(() => {
+                            return res.status(200).json({ success: true });
+                        })
+                        .catch(err => {
+                            return res
+                                .status(200)
+                                .json({ errors: [{ error: err }] });
+                        });
+                } else
+                    return res
+                        .status(404)
+                        .json({ errors: [{ password: 'incorrect' }] });
+            })
+            .catch(err => {
+                return res.status(404).json({ error: err });
```

Comment:
maybe you want errors here instead? just to stay consistent with the logic used in my above comment?
```suggestion
                return res.status(404).json({ errors: err });
```

Reply:
good catch